### PR TITLE
[dev-v2.9] rancher-vsphere-cpi 104.0.0+up1.8.0 add 

### PR DESCRIPTION
**Pull request by krunalhinguu**
### Updates

- Added new version `104.0.0+up1.8.0` for `rancher-vsphere-cpi` with commit hash `2d84611dada81f4987e67aae58c25a77392c956c`

### Related

- https://github.com/rancher/rancher/issues/45746

### CODE REVIEW

**Review by snasovich — CHANGES_REQUESTED** [review]
The same comment as https://github.com/rancher/charts/pull/4165#pullrequestreview-2154256596 applies (in principle).

**Comment by krunalhinguu** [comment]
@snasovich I have updated the code. let me if it is correct, i can do the same in another PR

**Review by snasovich — CHANGES_REQUESTED** [review]
Same comment (in principle) as https://github.com/rancher/charts/pull/4165#pullrequestreview-2156939721 applies

**Review by snasovich — APPROVED** [review]
Approving provided `index.yaml` generation is fixed

File: charts/rancher-vsphere-cpi/104.0.0+up1.8.0/Chart.yaml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+  catalog.cattle.io/kube-version: '>= 1.18.0-0 < 1.31.0-0'
   catalog.cattle.io/namespace: kube-system
   catalog.cattle.io/os: linux
   catalog.cattle.io/permits-os: linux,windows
-  catalog.cattle.io/rancher-version: '>= 2.9.0-0 < 2.10.0-0'
+  catalog.cattle.io/rancher-version: '>= 2.9.0-0'
   catalog.cattle.io/release-name: vsphere-cpi
 apiVersion: v1
-appVersion: 1.7.0
+appVersion: 1.30.1
```
Comment (jiaqiluo) [inline charts/rancher-vsphere-cpi/104.0.0+up1.8.0/Chart.yaml:11]: appVersion should be 1.8.0 matching the upstream app version

**Pull request merged**